### PR TITLE
refactor(hooks): #345 - Move API related types to separate package

### PR DESCRIPTION
**Pull request by grzesuav**
# Overview

This is first PR going to move forward with API versioning of hooks - as discussed in https://github.com/metacontroller/metacontroller/discussions/370 - for now extracting some of the api classes uses for serialization. Goal would be to isolate all classes being serialized in api versions, to allow handling more than one API version (which is needed for backward compatibility)

### CODE REVIEW

File: test/integration/decorator/decorator_test.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ package decorator
 
 import (
 	"context"
+	v12 "metacontroller/pkg/controller/common/customize/api/v1"
```
Comment (mjsmith1028) [inline test/integration/decorator/decorator_test.go:21]: should the name `v12` be something like `customizev1` ?

**Review by mjsmith1028 — DISMISSED** [review]
LGTM just minor comment/question

File: test/integration/decorator/decorator_test.go (hunk 1 above)
Comment (grzesuav) [reply test/integration/decorator/decorator_test.go:21]: oh, right, this was automatically added by intellij, will change it

File: pkg/controller/decorator/controller.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ package decorator
 import (
 	"context"
 	"fmt"
+	v12 "metacontroller/pkg/controller/decorator/api/v1"
```
Comment (AmitKumarDas) [inline pkg/controller/decorator/controller.go:22]: Is this typo? Why v12?

**Review by AmitKumarDas — DISMISSED** [review]
/LGTM
Left one minor comment w.r.t `v12` import alias

File: pkg/controller/decorator/controller.go (hunk 2 above)
Comment (grzesuav) [reply pkg/controller/decorator/controller.go]: yes, goland didi it as `v1` was taken, fixed

**Review by AmitKumarDas — APPROVED** [review]

**Pull request merged**